### ISSUE TITLE
一些关于训练时间的问题

### ISSUE THREAD

**Issue by baihy11111111** [rise-policy/RISE#7]
hi！非常感谢你的工作。我在训练时也遇到了加载时间过长的问题，我采集了50条数据，每条数据大概有150张图片。在训练过程中，我采用640*480的分辨率，但是每一个epoch还是要耗时5分钟，进度条卡在0%4分钟左右，我想请问你们在训练时遇到相同的问题了吗，如何进行解决？

**Comment by chenxi-wang** [comment]
你好，目前版本的点云处理会占用 dataloader 大部分时间，可能是 cpu 核数不够多，到瓶颈了。不过你用的数据量不是特别多，应该能在一分半内跑完才对。能麻烦分享一下你设置的 num_gpus, batch_size, num_workers 和机器的 cpu 核数吗？

另外，我们最近也在做 RISE 训练的优化，大概思路是第一个 epoch 保存 voxel downsample 后的点云到内存或写入硬盘（取决于内存容量），对于单视角点云和多视角点云速度提升都很明显。目前还在内部测试，没问题的话会更新上来，如果你需要的话也可以自己尝试一下，应该不麻烦

**Comment by baihy11111111** [comment]
好的 我使用的是
python train.py --data_path data/pick_place  --aug --num_action 20 --voxel_size 0.005 --obs_feature_dim 512 --hidden_dim 512 --nheads 8 --num_encoder_layers 4 --num_decoder_layers 1 --dim_feedforward 2048 --dropout 0.1 --ckpt_dir logs/pick_place_1 --batch_size 64 --num_epochs 2000 --save_epochs 100 --num_
workers 16 --seed 233

另外我用单卡H800，32个核

**Comment by chenxi-wang** [comment]
单卡需要设置 OMP_NUM_THREADS=1，如果没有其他程序吃 cpu 的话可以把 num_worker 调成 24 试试。对于 batch_size=64 的配置，最好把学习率同步调小，不然收敛速度会受影响，可以试试 1e-4 或者 5e-5

**Comment by baihy11111111** [comment]
单卡需要设置 OMP_NUM_THREADS=1，那么多卡需要设置参数是多少呢，看来我的主要影响就是这个参数

**Comment by baihy11111111** [comment]
好的 非常感谢你的解释
